server/controllers/userAuth.js

```diff
@@ -24,8 +33,102 @@ const sendDemoUser = (req, resp) => {
   resp.redirect('/');
 };
 
+const signUp = (req, resp, next) => {
+  const { email, password, passwordConfirm, username } = req.body;
+  const sanitizedEmail = sanitize(email);
+  const sanitizedPassword = sanitize(password);
+  const sanitizedPasswordConfirm = sanitize(passwordConfirm);
+  const sanitizedUsername = sanitize(username);
+  const errors = {};
+  const { isEmail, isEmpty, isLength, matches } = validator;
+
+  if (!isLength(sanitizedUsername, { min: 1, max: 32 })) {
+    errors.nameError = 'authorization.input.username.invalid';
+  }
+
+  if (isEmpty(sanitizedEmail) || !isEmail(sanitizedEmail)) {
+    errors.emailError = 'authorization.input.email.invalid';
+  }
+
+  if (!matches(sanitizedPassword, /^[^\s]{4,32}$/)) {
+    errors.passwordError = 'authorization.input.password.invalid';
+  }
+
+  if (sanitizedPassword !== sanitizedPasswordConfirm) {
+    errors.passwordConfirmError = 'authorization.input.password.not-match';
```

Comment:
are these localization ids? If yes than this shoud not be this way - in case we want to change the id on client we will need to update backend. This should be some string error which is mapped on client to specific localization id.